plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/domains/NullableDomain.kt

```diff
@@ -37,13 +37,15 @@ import org.jetbrains.kotlin.formver.viper.ast.Exp.Companion.Trigger1
 object NullableDomain : BuiltinDomain("Nullable") {
     val T = Type.TypeVar("T")
     override val typeVars: List<Type.TypeVar> = listOf(T)
-    val Nullable: Type = this.toType()
+
+    // Always use this instead of `toType` as it makes sure the type variables are mapped correctly.
+    fun nullableType(elemType: Type): Type.Domain = toType(mapOf(T to elemType))
 
     private val xVar = Var("x", T)
-    private val nxVar = Var("nx", Nullable)
+    private val nxVar = Var("nx", nullableType(T))
 
-    val nullFunc = createDomainFunc("null", emptyList(), Nullable)
-    val nullableOf = createDomainFunc("nullable_of", listOf(xVar.decl()), Nullable)
+    val nullFunc = createDomainFunc("null", emptyList(), nullableType(T))
+    val nullableOf = createDomainFunc("nullable_of", listOf(xVar.decl()), nullableType(T))
```

Comment:
Wasn't the plan to replace this with `cast`?

Reply:
Yes you're right, I forgot to change this in this version of the PR.